tox.ini

```diff
@@ -80,33 +80,37 @@ deps =
 setenv =
   TEST_JUJU_CHANNEL = 2.9/stable
 changedir = {toxinidir}/tests/functional
-commands = functest-run-suite
+commands = functest-run-suite {posargs:--keep-faulty-model}
 deps =
   git+https://github.com/openstack-charmers/zaza.git#egg=zaza
   -r {toxinidir}/tests/functional/requirements.txt
 
-[testenv:func33-jammy]
+[testenv:func31]
 setenv =
   TEST_JUJU_CHANNEL = 3.3/stable
 changedir = {toxinidir}/tests/functional
 commands =
-  # NOTE(mertkirpici): functest-run-suite -b jammy focal
-  # or functest-run-suite -b jammy -b focal does not work
-  # https://github.com/openstack-charmers/zaza/issues/614
-  functest-run-suite -b jammy
+  functest-run-suite {posargs:--keep-faulty-model}
 deps =
-  git+https://github.com/openstack-charmers/zaza.git@libjuju-3.1#egg=zaza[juju-33]
+  git+https://github.com/openstack-charmers/zaza.git@libjuju-3.1#egg=zaza[juju-31]
```

Comment:
where do you find the "extra" : `zaza[juju-31]` and also the following?

Trying to `pip install git+https://github.com/openstack-charmers/zaza.git@libjuju-3.1#egg=zaza[juju-32]`
but getting the following warning:
```
WARNING: zaza 0.0.2.dev1 does not provide the extra 'juju-32'
```

Reply:
even though I change the "extra", but it will always install `juju-3.1.2.0.dist-info/`